File: Array&String/String/CPP/make_the_string_great.cpp
#include <string>
#include <cmath>

using namespace std;

class Solution {
public:
    string makeGood(string s) {
        string ans;
        ans += s[0];
        int n = s.size();
        for (int i = 1; i < n; i++) {
            char cur = s[i];
            if (!ans.empty() && abs(static_cast<int>(ans.back()) - static_cast<int>(cur)) == 32) {
                ans.pop_back();
                continue;
            }
            ans += cur;
        }
        return ans;
    }
};

File: Array&String/String/Python/make_the_string_great.py
class Solution:
    def makeGood(self, s: str) -> str:
        n, res = len(s), ""
        for c in s:
            if len(res) > 0 and abs(ord(res[-1]) - ord(c)) == 32: res = res[:-1]
            else: res += c
        return res